Apps/W1/HybridBaseDeployment/app/src/pages/CompanyMigrationStatus.Page.al

```diff
@@ -0,0 +1,82 @@
+namespace Microsoft.DataMigration;
+
+page 40066 "Company Migration Status"
+{
+    ApplicationArea = All;
+    Caption = 'Company Migration Status';
+    PageType = List;
+    SourceTable = "Hybrid Company Status";
+    UsageCategory = None;
+    InsertAllowed = false;
+    Editable = false;
+    DeleteAllowed = false;
+
+    layout
+    {
+        area(Content)
+        {
+            repeater(General)
+            {
+                field(Name; Rec.Name)
+                {
+                    ToolTip = 'Specifies the value of the Name field.';
+                }
+                field("Upgrade Status"; Rec."Upgrade Status")
+                {
+                    ToolTip = 'Specifies the value of the Upgrade Status field.';
+                }
+                field(Validated; Rec.Validated)
+                {
+                    ToolTip = 'Specifies the value of the Validated field.';
+                }
+            }
+        }
+    }
+
+    actions
+    {
+        area(Promoted)
+        {
+            actionref(RunAllValidation_Promoted; RunAllValidation)
+            {
+            }
+        }
+
+        area(Processing)
+        {
+            action(RunAllValidation)
+            {
+                ApplicationArea = All;
+                Caption = 'Run All Validation';
+                ToolTip = 'Run validation on all migrated companies that havve yet to be validated.';
+                Image = Process;
+
+                trigger OnAction()
+                var
+                    IntelligentCloudSetup: Record "Intelligent Cloud Setup";
+                    HybridCompanyStatus: Record "Hybrid Company Status";
+                    MigrationValidationMgmt: Codeunit "Migration Validation Mgmt.";
+                    SessionsStarted: Boolean;
+                begin
+                    if not IntelligentCloudSetup.Get() then
+                        exit;
+
+                    HybridCompanyStatus.SetFilter(Name, '<>%1', '');
```

Comment:
I believe we have some code duplication here, it would be rather good if we could consolidate